esplora/src/main/scala/org/bitcoins/esplora/EsploraClient.scala

```diff
@@ -0,0 +1,226 @@
+    val url = baseUrl + s"/address/$addr/txs/mempool"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequestAndParse[Vector[EsploraTransaction]](request)
+  }
+
+  def getBlock(hash: DoubleSha256DigestBE): Future[EsploraBlock] = {
+    val url = baseUrl + s"/block/${hash.hex}"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequestAndParse[EsploraBlock](request)
+  }
+
+  def getBlockHeader(hash: DoubleSha256DigestBE): Future[BlockHeader] = {
+    val url = baseUrl + s"/block/${hash.hex}/header"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequest(request).map(BlockHeader.fromHex)
+  }
+
+  def getBlockHashAtHeight(height: Int): Future[DoubleSha256DigestBE] = {
+    val url = baseUrl + s"/block-height/$height"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequest(request).map(DoubleSha256DigestBE.fromHex)
+  }
+
+  def getRawBlock(hash: DoubleSha256DigestBE): Future[Block] = {
+    val url = baseUrl + s"/block/${hash.hex}/raw"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRawRequest(request).map { byteString =>
+      val bytes = ByteVector(byteString.toArray)
+      Block.fromBytes(bytes)
+    }
+  }
+
+  // -- ChainQueryApi --
+  override def getBestHashBlockHeight()(implicit
+      _ec: ExecutionContext): Future[Int] = {
+    val url = baseUrl + s"/blocks/tip/height"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequest(request).map(_.toInt)(ec)
+  }
+
+  override def getBestBlockHash(): Future[DoubleSha256DigestBE] = {
+    val url = baseUrl + s"/blocks/tip/hash"
+    val request = HttpRequest(uri = url, method = HttpMethods.GET)
+
+    sendRequest(request).map(DoubleSha256DigestBE.fromHex)
+  }
+
+  override def getBlockHeight(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    getBlock(blockHash)
+      .map(_.height)
+      .map(Some(_))
+      .recover { case NonFatal(_) => None }
+  }
+
+  override def getNumberOfConfirmations(
+      blockHash: DoubleSha256DigestBE): Future[Option[Int]] = {
+    val tipF = getBestHashBlockHeight()
+    val heightF = getBlockHeight(blockHash)
+
+    for {
+      tip <- tipF
+      heightOpt <- heightF
+    } yield heightOpt.map(height => tip - height + 1)
+  }
+
+  override def getFilterCount(): Future[Int] = Future.failed(
+    new UnsupportedOperationException("Esplora does not support block filters"))
+
+  override def getHeightByBlockStamp(blockStamp: BlockStamp): Future[Int] = {
+    blockStamp match {
+      case blockHeight: BlockStamp.BlockHeight =>
+        Future.successful(blockHeight.height)
+      case blockHash: BlockStamp.BlockHash =>
+        getBlock(blockHash.hash).map(_.height)
+      case blockTime: BlockStamp.BlockTime =>
+        Future.failed(
+          new UnsupportedOperationException(s"Not implemented: $blockTime"))
```

Comment:
It would be useful to include `Esplora` in the error message so it is easy to see that the `ChainQueryApi` being used is `Esplora` like you did in the `getFilterCount()` message.